app/components/user_programme_course_bookings_with_asides_component.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class UserProgrammeCourseBookingsWithAsidesComponent < CmsWithAsidesComponent
+  delegate :current_user, to: :helpers
+
+  def initialize(programme:, aside_slug: nil)
+    aside_sections = if aside_slug.nil?
+      nil
+    else
+      [{slug: aside_slug}]
+    end
+
+    super(aside_sections:)
+    @programme = programme
+  end
+
+  def before_render
+    @current_user = current_user
+  end
+
+  private
+
+  def course_sections
+    [{title: "Booked courses", courses: in_progress_courses},
+      {title: "Completed courses", courses: completed_courses}]
+  end
+
+  def in_progress_courses
+    @current_user.achievements
+      .in_state(:in_progress, :enrolled)
+      .with_courses.order("created_at DESC")
+      .belonging_to_programme(@programme)
+  end
+
+  def completed_courses
+    @current_user.achievements
+      .in_state(:complete)
+      .with_courses.order("created_at DESC")
+      .belonging_to_programme(@programme)
+  end
+
+  def course_icon_class(activity)
```

Comment:
We don't need to redefine this, as we have it in a helper method, in fact we seem have multiple versions of it 
https://github.com/NCCE/teachcomputing.org/blob/d811cae9eea9258b60829ea757d08eac01de6569/app/helpers/activities_helper.rb#L13
https://github.com/NCCE/teachcomputing.org/blob/d811cae9eea9258b60829ea757d08eac01de6569/app/helpers/dashboard_helper.rb#L10